src/include/sof/trace/trace.h

```diff
@@ -264,6 +256,26 @@ _thrown_from_macro_BASE_LOG_in_trace_h
  * linker section and replaced by a &log_entry pointer to it. This must
  * be a macro for the source location to be meaningful.
  */
+#ifndef __ZEPHYR__
+#define _log_message(log_func, atomic, lvl, comp_class, ctx, id_1, id_2, format, ...)	\
+do {											\
+	_DECLARE_LOG_ENTRY(lvl, format, comp_class,					\
+			   META_COUNT_VARAGS_BEFORE_COMPILE(__VA_ARGS__));		\
+	STATIC_ASSERT_ARG_SIZE(__VA_ARGS__);						\
+	STATIC_ASSERT(_TRACE_EVENT_MAX_ARGUMENT_COUNT >=				\
+			META_COUNT_VARAGS_BEFORE_COMPILE(__VA_ARGS__),			\
+		BASE_LOG_ASSERT_FAIL_MSG						\
+	);										\
+	(lvl <= MTRACE_DUPLICATION_LEVEL ? \
+		mtrace_dict_entry(true, (uint32_t)&log_entry, \
+				  META_COUNT_VARAGS_BEFORE_COMPILE(__VA_ARGS__), \
+				  ##__VA_ARGS__) : 0); \
```

Comment:
The `?:` operator is (usually) used for returning a value.  Do I understand it correctly that you just meant an `if` here? Also it would be nice to keep indentation for end-of-line backslashes